CMSgov/mdct-mcr
# MDCT-1993 api updates

## Description
[MDCT-1993](https://qmacbis.atlassian.net/browse/MDCT-1993) modify api to match standards
1. Fetch to a specific item returns a 404 when the response is empty (instead of 200)
2. Create calls return a 201 instead of 200
3. Fetch’s with full primary keys will convert to a get from a query
4. An update where the matching record does not exist returns a 404 (instead of a 500)

### How to test
Try out the [mcr postman collection from confluence](https://qmacbis.atlassian.net/l/cp/Me1mwh6p)! Check that:
- Create now returns a 201 instead of 200
- Update without existing item returns 404
- Update without new data returns 400
- Fetch where specific item doesn't exist returns 404 (does not apply to queries like "reportsByState")

ping me if you want help!

### Changed Dependencies
none

## Code author checklist

## Reviewer checklist (two different people)
